docs/basic-features/environment-variables.md

```diff
@@ -124,3 +124,9 @@ This one is useful when running tests with tools like `jest` or `cypress` where
 There is a small difference between `test` environment, and both `development` and `production` that you need to bear in mind: `.env.local` won't be loaded, as you expect tests to produce the same results for everyone. This way every test execution will use same env defaults across different executions by ignoring your `.env.local` (which is intended to override the default set).
 
 > **Note**: similar to Default Environment Variables, `.env.test` file should be included in your repository, but `.env.test.local` shouldn't, as `.env*.local` are intended to be ignored through `.gitignore`.
+
+## NODE_ENV
+
+The value of `NODE_ENV` is used to determined which `.env.*` file to load when running the application. However, this value is not passed to your application code when using the `next` CLI. Instead, when running `next dev` then `process.env.NODE_ENV` is set to `development`, and when running `next start` it is set to `production`.
```

Comment:
`NODE_ENV` should never be manually set, so I don't think this is warranted advice. Maybe just clarify that it's `development` for `next dev` and `production` for `next build`?

Reply:
Yeah that's probably fair. I just wanted to decouple from the problem I was personally seeing which was, when running `next build` is run inside of a jest context (using jest-puppeteer) it will fetch the `.env.test` config but run with `NODE_ENV=production`. That said, I think just knowing the `environments` for `next build` and `next dev` would be enough like you say 👍